app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderPresenter.kt

```diff
@@ -168,8 +169,13 @@ class ReaderPresenter(
         val currentChapters = viewerChaptersRelay.value
         if (currentChapters != null) {
             currentChapters.unref()
-            saveChapterProgress(currentChapters.currChapter)
-            saveChapterHistory(currentChapters.currChapter)
+            val currentChapter = currentChapters.currChapter
+            saveChapterProgress(currentChapter)
+            saveChapterHistory(currentChapter)
+            val currentChapterPageCount = currentChapter.pages?.size ?: 1
+            if (currentChapter.chapter.last_page_read.toDouble() / currentChapterPageCount > 0.2) {
```

Comment:
i'd say 0.33 would be a better but its a random value all the same really 
also add a check if the currentChapters.previousChapter was downloaded and if it was or this chapter has progressed enough, then downloadNextChapters (also pass a bool stating if the last chapter was downloaded to downloadNextChapters)